### ISSUE TITLE
Failed at visualize stage to generate the .hic file. Out of memory issues.

### ISSUE THREAD

**Issue by Zahnhh** [theaidenlab/3d-dna#152]
When I deal with a huge merged_nodups.txt file about 800G. There is an issue like '.Exception in thread "main" java.lang.OutOfMemoryError: GC overhead limit exceeded'. It happened at the visualize stage when generating the .hic file.
So I only reserved several lower resolutions (-r 2500000,1000000,500000,250000,100000) and raise my node memory to 1920G. But it didn't work. 
Do you have any advice on how to fix this?

**Comment by Zahnhh** [comment]
And when I check the task, I found 'Memory Utilized: 50.01 GB Memory Efficiency: 2.22% of 2.20 TB'

**Comment by dudcha** [comment]
Since you are writing this as part of the 3d-dna thread I am assuming you are trying to visualize a draft assembly. Are you following the instructions on page 5 of the genome assembly book? Dnazoo.org/methods? -Olga

**Comment by Zahnhh** [comment]
> Since you are writing this as part of the 3d-dna thread I am assuming you are trying to visualize a draft assembly. Are you following the instructions on page 5 of the genome assembly book? Dnazoo.org/methods? -Olga
> […](#)
> On Jul 8, 2022, at 5:42 AM, Zahnhh ***@***.***> wrote: ﻿ And when I check the task, I found 'Memory Utilized: 50.01 GB Memory Efficiency: 2.22% of 2.20 TB' — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.

**Comment by Zahnhh** [comment]
> 

Yes, I actually follow the instructions. But this problem is related to some bugs in the script of juicerbox.

**Comment by Zahnhh** [comment]
> Since you are writing this as part of the 3d-dna thread I am assuming you are trying to visualize a draft assembly. Are you following the instructions on page 5 of the genome assembly book? Dnazoo.org/methods? -Olga
> […](#)
> On Jul 8, 2022, at 5:42 AM, Zahnhh ***@***.***> wrote: ﻿ And when I check the task, I found 'Memory Utilized: 50.01 GB Memory Efficiency: 2.22% of 2.20 TB' — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.

Yes, I actually follow the instructions. But this problem is related to some bugs in the script of juicerbox.

**Comment by Zahnhh** [comment]
I fixed this issue by tuning the parameter -xmx in juicer_tool.sh and requesting more memory. But I forget to change the resolution of the splitter, polisher and editor. So it's still a default which is not contained in my contact map resolution. I have no idea if it's gonna influence my final result, but my contact map seems fine.

**Comment by conchoecia** [comment]
Just came here to say thank you for this suggestion, @Zahnhh. It would be nice if there were an option to specify the max amount of RAM to use at runtime rather than having it be hard-coded into `juicer_tool.sh`

**Comment by zhangna123-ux** [comment]
> I fixed this issue by tuning the parameter -xmx in juicer_tool.sh and requesting more memory. But I forget to change the resolution of the splitter, polisher and editor. So it's still a default which is not contained in my contact map resolution. I have no idea if it's gonna influence my final result, but my contact map seems fine.

Hi, Zahnhh, I would like to ask you about the value that you set for tuning the parameter -Xmx in juicer_tool.sh and how to request more memory when resolving this issue. I am facing the same error now and my merged_nodups.txt file size is 511G. What would be a suitable value should I set for the parameter -Xms? Here is the memory status of my server:
                    total          used         free        shared       buff/cache       available
Mem:         629Gi         29Gi       293Gi       127Mi           306Gi             596Gi
Swap:           0B             0B            0B
I'm looking forward to your reply and thank you very much!

**Comment by zhangna123-ux** [comment]
> I fixed this issue by tuning the parameter -xmx in juicer_tool.sh and requesting more memory. But I forget to change the resolution of the splitter, polisher and editor. So it's still a default which is not contained in my contact map resolution. I have no idea if it's gonna influence my final result, but my contact map seems fine.

Hi, Zahnhh, I would like to ask you about the value that you set for tuning the parameter -Xmx in juicer_tool.sh and how to request more memory when resolving this issue. I am facing the same error now and my merged_nodups.txt file size is 511G. What would be a suitable value should I set for the parameter -Xms? Here is the memory status of my server:
                  total      used       free     shared     buff/cache     available
Mem:       629Gi     29Gi     293Gi     127Mi          306Gi            596Gi
Swap:         0B          0B          0B
I'm looking forward to your reply and thank you very much!

**Comment by zhangna123-ux** [comment]
> I fixed this issue by tuning the parameter -xmx in juicer_tool.sh and requesting more memory. But I forget to change the resolution of the splitter, polisher and editor. So it's still a default which is not contained in my contact map resolution. I have no idea if it's gonna influence my final result, but my contact map seems fine.

Hi, @Zahnhh, I would like to ask you about the value that you set for tuning the parameter -Xmx in juicer_tool.sh and how to request more memory when resolving this issue. I am facing the same error now and my merged_nodups.txt file size is 511G. What would be a suitable value should I set for the parameter -Xmx? Here is the memory status of my server:
![4491704805441_ pic](https://github.com/aidenlab/3d-dna/assets/127749163/2200f41f-d745-47ba-9278-1a88a42da8c8)

I'm looking forward to your reply and thank you very much!
